esp8266-fastled-webserver/data/index.htm

```diff
@@ -7,12 +7,45 @@
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <title>ESP8266 + FastLED</title>
 
-  <!-- request CSS from internet CDN -->
-  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3" crossorigin="anonymous">
-  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jquery-minicolors/2.3.6/jquery.minicolors.min.css" integrity="sha512-BVeRnUOL0G7d4gXmj+0VxpoiQuEibKQtlkclADKvCdNrESs0LA6+H8s1lU455VqWFtHBfF/pKDGw/CMat2hqOg==" crossorigin="anonymous" referrerpolicy="no-referrer" />
-  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.7.2/font/bootstrap-icons.min.css">
+  <!-- Fallback to load resources from FileSystem if not available online -->
+  <script>
+    function CssJsFallback(el) {
+      if (el instanceof HTMLScriptElement || el instanceof HTMLLinkElement) {
+        let Url = el.getAttribute('data-fallback');
+        if (Url) {
```

Comment:
This appears to modify the attributes of the element that fails to load via use of the `onerror()` callback.

Question:
Is there a specification that requires the element to be re-loaded when its attributes are changed in this callback?

Reply:
[https://www.w3schools.com/jsref/event_onerror.asp](https://www.w3schools.com/jsref/event_onerror.asp) is the specification for `onerror()` compatible to "all" common browsers.

There is no direct reload neccessary. At least in chrome it just reloads the ressource from the fallback path.